src/mlpack/methods/ann/layer/star_relu.hpp

```diff
@@ -0,0 +1,109 @@
+/**
+ * @file methods/ann/layer/star_relu.hpp
+ * @author Mayank Raj
+ *
+ * Definition of StarReLU layer introduced by Yu et al.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_STAR_RELU_HPP
+#define MLPACK_METHODS_ANN_LAYER_STAR_RELU_HPP
+
+#include <mlpack/prereqs.hpp>
+
+#include "layer.hpp"
+
+namespace mlpack {
+
+/**
+ * The StarReLU activation function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) &=& s \cdot (\text{ReLU}(x))^2 + b
+ * f'(x) &=& s \cdot 2 \cdot \text{ReLU}(x)
+ * @f}
```

Comment:
Can you cite the paper (https://arxiv.org/abs/2210.13452v2) as well, see https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/activation_functions/hard_swish_function.hpp for an example.